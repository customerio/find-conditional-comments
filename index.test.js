const findConditionalComments = require(".");

test("find conditional comments", () => {
  const comments = findConditionalComments(`
    Basic  
    <!--[if mso]><![endif]-->

    Basic multi-line
    <!--[if MAC]>
      Multi-line
    <![endif]-->

    Great than
    <!--[if gt mso 14]> Everything above Outlook 2010 <![endif]-->
    
    Less than
    <!--[if lt mso 14]> Everything below Outlook 2010 <![endif]-->
    
    Greater than or equal
    <!--[if gte mso 14]> Outlook 2010 and above <![endif]-->
    
    Less than or equal
    <!--[if lte mso 14]> Outlook 2010 and below <![endif]-->
    
    Or statement
    <!--[if (mso 12)|(mso 16)]> Outlook 2007 / 2016 only <![endif]-->
    
    And statement
    <!--[if (mso 12)&(mso 16)]> Outlook 2007 / 2016 only <![endif]-->
    
    Not statement
    <!--[if !mso]><!--> All Outlooks will ignore this <!--<![endif]-->

    Alt-syntax
    <![if mso]><![endif]>


    Not these folks. They look like they could match but don't 
    really work for some reasons.

    <!-- regular HTML comment -->
    <!--[if ]><!--> No condition <!--<![endif]-->
    <!--[if mso]
    >
      The condition is broken with a new line
    <![endif]-->
  `);

  expect(comments).toMatchSnapshot();
});
