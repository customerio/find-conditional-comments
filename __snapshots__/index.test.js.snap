// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find conditional comments 1`] = `
Array [
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if mso]>",
    "range": Array [
      17,
      42,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if MAC]>",
    "range": Array [
      69,
      116,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if gt mso 14]>",
    "range": Array [
      137,
      199,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if lt mso 14]>",
    "range": Array [
      223,
      285,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if gte mso 14]>",
    "range": Array [
      321,
      377,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if lte mso 14]>",
    "range": Array [
      410,
      466,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if (mso 12)|(mso 16)]>",
    "range": Array [
      493,
      558,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]-->",
    "downlevel": "hidden",
    "isComment": true,
    "open": "<!--[if (mso 12)&(mso 16)]>",
    "range": Array [
      586,
      651,
    ],
  },
  Object {
    "bubble": true,
    "close": "<!--<![endif]-->",
    "downlevel": "revealed",
    "isComment": true,
    "open": "<!--[if !mso]><!-->",
    "range": Array [
      679,
      745,
    ],
  },
  Object {
    "bubble": false,
    "close": "<![endif]>",
    "downlevel": "revealed",
    "isComment": false,
    "open": "<![if mso]>",
    "range": Array [
      766,
      787,
    ],
  },
]
`;
